bluepyemodel/icselector/modules/mechanism.py

```diff
@@ -0,0 +1,163 @@
+"""Mechanism class corresponding to mechanisms fields in the icmapping file."""
+
+from pprint import pprint
+
+import numpy as np
+
+from .distribution import Distribution
+from .distribution import asdict
+
+
+class Mechanism:
+    """Holds all collected information associated with a Neuron mechanism.
+    Corresponds to mechanisms fields in the icmapping file, with added
+    fields containing additional information from GeneSelector."""
+
+    def __init__(self, **kwargs):
+        self.nexus = {}
+        self.model = {}
+        self.requires = []
+        self.status = None
+        self.distribution = Distribution()
+        self._mapped_from = []
+        self._bounds = {}
+        self._selected = False
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        # Some initial values
+        if "parameters" in self.model:
+            for param in self.model["parameters"]:
+                self.set_parameters(**{param: 0})
+        self.set_gbar([0, 1])
+
+    def select(self, check_status=None):
+        """Select mechanism for inclusion in model configuration.
+        Args:
+            check_status (str): only select the mechanism if status meets criterium
+        """
+
+        if not self.status:
+            self._selected = True
+            return
+
+        if check_status == "stable":
+            self._selected = check_status == self.status
+        else:
+            self._selected = True
+
+    def deselect(self):
+        """Deselect mechanism."""
+
+        self._selected = False
+
+    def is_selected(self):
+        """Mechanism has been selected for inclusion in model configuration.
+        Args:
```

Comment:
There is no args in this function, can remove them from docstring

Reply:
Done in [942a287](https://github.com/BlueBrain/BluePyEModel/pull/144/commits/942a287eb332cb58b5608511616eefcd2e76e880)